src/NLog/LogFactory.cs

```diff
@@ -1092,15 +1093,51 @@ public void ResetCandidateConfigFilePath()
         private static IEnumerable<string> GetDefaultCandidateConfigFilePaths()
         {
             // NLog.config from application directory
-            if (CurrentAppDomain?.BaseDirectory != null)
+            string baseDirectory = PathHelpers.TrimDirectorySeparators(CurrentAppDomain?.BaseDirectory);
+            if (!string.IsNullOrEmpty(baseDirectory))
+                yield return Path.Combine(baseDirectory, "NLog.config");
+
+            bool platformFileSystemCaseInsensitive = PlatformDetector.IsWin32;
+            if (!platformFileSystemCaseInsensitive && !string.IsNullOrEmpty(baseDirectory))
```

Comment:
PS: windows could be case sensitive, but highly unlikely ;)

Reply:
Would cause most applications on Windows to fail badly, which is why I skip the extra file-checks